mypyc/test-data/run.test

```diff
@@ -1065,6 +1065,36 @@ assert d == dict(object.__dict__)
 
 assert u(10) == 10
 
+[case testDictIterationMethodsRun]
+from typing import Dict
+def print_dict_methods(d1: Dict[int, int],
+                       d2: Dict[int, int],
+                       d3: Dict[int, int]) -> None:
+    for k in d1.keys():
+        print(k)
+    for k, v in d2.items():
+        print(k)
+        print(v)
+    for v in d3.values():
+        print(v)
+
+class Custom(Dict[int, int]): pass
+[file driver.py]
+from native import print_dict_methods, Custom
+print_dict_methods({}, {}, {})
+print_dict_methods({1: 2}, {3: 4}, {5: 6})
```

Comment:
Also test dictionaries with multiple items.

Maybe test what happens if we change the length of the dictionary during iteration, such as by clearing it?

Reply:
> Maybe test what happens if we change the length of the dictionary during iteration, such as by clearing it?

There is an existing (huge) test that changes dict size and verifies that `RuntimeError` is raised. I can add a small dedicated test case, since this is probably important.